lib/app/service/app.go

```diff
@@ -828,7 +820,10 @@ func (r *applications) resolveManifest(manifestBytes []byte) (*schema.Manifest,
 	}
 
 	message := "Dependency %v excluded from manifest"
-	manifest.Dependencies.Apps = appservice.Wrap(loc.Filter(appservice.Unwrap(manifest.Dependencies.Apps), r.Config.ExcludeDeps, message))
+	manifest.Dependencies.Apps = appservice.Wrap(
+		loc.Filter(appservice.Unwrap(manifest.Dependencies.Apps),
+			appservice.AppsToExclude(*manifest), message),
```

Comment:
See comment above - instead of using configuration which is not relayed everywhere, compute the list of excluded applications where necessary.